xapian-core/tests/api_sorting.cc

```diff
@@ -321,79 +321,77 @@ DEFINE_TESTCASE(sortfunctorremote1, remote) {
 }
 
 DEFINE_TESTCASE(replace_weights, backend) {
-    // open the database (in this case a simple text file
-    // we prepared earlier)
 
     Xapian::Database mydb(get_database("apitest_onedoc"));
 
     Xapian::Enquire enquire(mydb);
 
-    // make a simple query, with one word in it - "word".
     enquire.set_query(Xapian::Query("word"));
 
-    // retrieve the top ten results (we only expect one)
+    // retrieve the top ten results (we only expect one).
     Xapian::MSet mymset = enquire.get_mset(0, 10);
 
-    // assigning new weight of -1 to the document.
+    /* assigning new weight of -1 to the document.
+     * a negative weight was chose to avoid matching the positive weight returned by enquire.
```

Comment:
No, I'd use a strictly positive weight. You might want to figure out what the actual weight of the single MSet item is in this case and just take a little off that. (It won't change unless we change the default weighting scheme, which is incredibly unlikely.)